packages/xr/src/lib/hooks/use-hand-joint.ts

```diff
@@ -0,0 +1,43 @@
+import { onDestroy } from 'svelte'
+import { currentWritable } from '@threlte/core'
+import type { HandJoints } from '../lib/hand-joints'
+import type * as THREE from 'three'
+import { useHand } from './use-hand'
+
+const onHandReady = (
+  handedness: 'left' | 'right',
+  joint: HandJoints,
+  callback: (jointSpace: THREE.XRJointSpace) => void
+) => {
+  const xrhand = useHand(handedness)
+
+  let id: number
+
+  const tick = () => {
+    const jointSpace = xrhand.current?.hand.joints[joint]
+
+    if (jointSpace?.jointRadius === undefined) {
+      id = window.setTimeout(tick, 100)
```

Comment:
is there a reason to use a timeout instead of a `useFrame`?

Reply:
Going to see if I can rework this a bit in this PR - unfortunately joints are assigned asynchronously and it doesn't seem like there's an event that three.js provides for when joints become available (they aren't immediately available when the hand loads). I added this polling function just because I didn't want to needlessly check every frame but maybe I was over-optimizing...

Reply:
The approach was inspired by: https://codesandbox.io/s/tp97r

Reply:
I added the check in `useFrame`. I guess that at most the hook is used in about 40 iterations (both hands, all joints) and 40 checks for properties are cheap.